javascript/index.js

```diff
@@ -4,37 +4,113 @@
 // 🚨🚨🚨 Comment out the below code before you start working on the code
 
 // Out of sync
-  getInstruction("mashedPotatoes", 0, (step1) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step1}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 1, (step2) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step2}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 2, (step3) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step3}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 3, (step4) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step4}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 4, (step5) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step5}</li>`;
-    document.querySelector("#mashedPotatoesImg").removeAttribute("hidden");
-  }, (error) => console.log(error));
-
 
 
 // Iteration 1 - using callbacks
-// ...
+
+// CALLBACK HELL
+
+  getInstruction("mashedPotatoes", 0, (step1) => {
+    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step1}</li>`;
+    getInstruction("mashedPotatoes", 1, (step2) => {
+      document.querySelector("#mashedPotatoes").innerHTML += `<li>${step2}</li>`;
+      getInstruction("mashedPotatoes", 2, (step3) => {
+        document.querySelector("#mashedPotatoes").innerHTML += `<li>${step3}</li>`;
+        getInstruction("mashedPotatoes", 3, (step4) => {
+          document.querySelector("#mashedPotatoes").innerHTML += `<li>${step4}</li>`;
+          getInstruction("mashedPotatoes", 4, (step5) => {
+            document.querySelector("#mashedPotatoes").innerHTML += `<li>${step5}</li>`;
+            document.querySelector("#mashedPotatoes").innerHTML += `<li>Mashed potatoes are ready!</li>`
+            document.getElementById("mashedPotatoesImg").removeAttribute("hidden");
+          });
+        });
+      });
+    });
+  },(error) => console.log(error));
+
 
 // Iteration 2 - using promises
-// ...
+obtainInstruction('steak', 0)
 
+  .then( (step0) => {
+    document.querySelector("#steak").innerHTML += `<li>${step0}</li>`
+    return steak[1];
```

Comment:
on each .then, you should return the obtainInstruction function with the food name and step, like:

```
return obtainInstruction("steak", 1);
return obtainInstruction("steak", 2);
return obtainInstruction("steak", 3);
(...)
```